backend/internal/service/handler/spotify/import_recommendations.go

```diff
@@ -0,0 +1,411 @@
+package spotify
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"platnm/internal/models"
+	"platnm/internal/service/ctxt"
+	"platnm/internal/storage"
+	"sync"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/zmb3/spotify/v2"
+)
+
+type importRecommendationsRequest struct {
+	Artists []string `json:"artists"`
+	Tracks  []string `json:"tracks"`
+	Genres  []string `json:"genres"`
+}
+
+type importRecommendationsResponse struct {
+	Artists      []models.Artist           `json:"artists"`
+	Tracks       []models.Track            `json:"tracks"`
+	Albums       []models.Album            `json:"albums"`
+	TrackArtists []bridgeTrackArtistResult `json:"track_artists"`
+	AlbumArtists []bridgeAlbumArtistResult `json:"album_artists"`
+}
+
+type bridgeTrackArtistResult struct {
+	TrackID  int `json:"track_id"`
+	ArtistID int `json:"artist_id"`
+}
+
+type bridgeAlbumArtistResult struct {
+	AlbumID  int `json:"album_id"`
+	ArtistID int `json:"artist_id"`
+}
+
+func (h *SpotifyHandler) ImportRecommendations(c *fiber.Ctx) error {
+	var req importRecommendationsRequest
+	if err := c.BodyParser(&req); err != nil {
+		return err
+	}
+
+	client, err := ctxt.GetSpotifyClient(c)
+	if err != nil {
+		return err
+	}
+
+	seeds := req.toSeeds()
+	recommendations, err := client.GetRecommendations(c.Context(), seeds, nil)
+	if err != nil {
+		return err
+	}
+
+	resp := handleRecommendations(c.Context(), recommendations, h.mediaRepository)
+	return c.Status(fiber.StatusOK).JSON(resp)
+}
+
+func handleRecommendations(ctx context.Context, recommendations *spotify.Recommendations, mr storage.MediaRepository) importRecommendationsResponse {
```

Comment:
i would use the [errgroup package](https://pkg.go.dev/golang.org/x/sync/errgroup) for this. will clean up and for this mvp i assume you want to fail on the first err, not chain/retry stage when err

Reply:
refer above comment